### ISSUE TITLE
assigning __init_subclass__ to an existing function

### ISSUE THREAD

**Issue by multimeric** [microsoft/pyright#5500]
**Describe the bug**
When assigning magic class methods like `__init_subclass__` to an existing function, and then subclassing that class, pyright erroneously says `error: Argument missing for parameter "cls" (reportGeneralTypeIssues)`.

**To Reproduce**
Run pyright on the following:
```python
def func(cls):
    pass

class Parent:
    __init_subclass__ = func

class Child(Parent):
    pass
```

**Expected behavior**
I expect no type errors in the above scenario. I can't find a specific justification for my claim, but I believe the approach to assigning this method is equivalent to the more normal version:

```python
class Parent:
    def __init_subclass__(cls):
        pass

class Child(Parent):
    pass
```

**Comment by erictraut** [comment]
Someone thought it would be a good idea to make `__init_subclass__` implicitly a class method even if it's not defined as such. It's odd special cases like this in Python that make it so difficult to get type checking correct in all cases. Not surprisingly, mypy generates the same error in this case. Grumble grumble.

I've added more special-case logic to avoid the type checking error. This will be included in the next release of pyright.

**Comment by multimeric** [comment]
I understand this is a mega edge case, thanks for the fix!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.318, which I just published. It will also be included in a future release of pylance.
